pkg/cassandra/config/schema.go

```diff
@@ -0,0 +1,147 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+import (
+	"bytes"
+	"embed"
+	"errors"
+	"fmt"
+	"text/template"
+	"time"
+
+	"github.com/jaegertracing/jaeger/pkg/cassandra"
+)
+
+//go:embed v004-go-tmpl.cql.tmpl
+var schemaFile embed.FS
+
+type TemplateParams struct {
+	Schema
+	// Replication is the replication strategy used
+	Replication string `mapstructure:"replication" valid:"optional"`
+	// CompactionWindowInMinutes is constructed from CompactionWindow for using in template
+	CompactionWindowInMinutes int64 `mapstructure:"compaction_window_size"`
+	// CompactionWindowUnit is the time unit of the CompactionWindow. Extracted from CompactionWindow
+	CompactionWindowUnit string `mapstructure:"compaction_window_unit"`
```

Comment:
if the previous value is always in minutes it doesn't seem like you need this one as a template parameter

Reply:
another alternative you can rename CompactionWindowInMinutes -> CompactionWindow, then you would use both parameters but the knowledge that it's always in minutes is going to be implementation detail in this file

Reply:
Removed `CompactionWindowUnit`